### PR TITLE
feat(Table): Add onRowClick event

### PR DESCRIPTION
**Pull request by ghost**
## Overview
Add an `onRowClick` event to the Table component

## Reason
Allows component consumers to react to clicks on items in the table

## Work carried out
- [x] New optional prop for `onRowClick` added to `TableProps`
- [x] Update tests

### CODE REVIEW

File: packages/react-component-library/src/components/Table/Table.tsx [hunk 1]
```diff
@@ -26,6 +26,11 @@ export interface TableProps extends ComponentWithClass {
    * Optional text caption to display above the component.
    */
   caption?: string
+  /**
+   * Optional, callback when a row is clicked
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/Table/Table.tsx]:
```suggestion
   * Optional handler invoked when a row is clicked.
```

Comment (m7kvqbe1) [reply packages/react-component-library/src/components/Table/Table.tsx]: Sorry I know this is really nitpicky. 

Just trying to align it with style of other prop interface comments (tone of voice / full stop).

File: packages/react-component-library/src/components/Table/Table.tsx [hunk 2]
```diff
@@ -61,6 +75,7 @@ export const Table: React.FC<TableProps> = ({
         <tbody>
           {tableData.map((row: RowProps) => (
             <StyledTableRow
+              onClick={() => handleRowClick(row.id)}
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/Table/Table.tsx:78]: Might be able to do this inline with optional chaining `onRowClick?(id)`.

Fine as is though.

**Review by m7kvqbe1 — COMMENTED** [review]
LGTM just change to the interface comment. 

Only nitpicky about these because they end up in the Storybook docs.

File: packages/react-component-library/src/components/Table/Table.tsx (hunk 1 above)
Comment (ghost) [reply packages/react-component-library/src/components/Table/Table.tsx]: 👍 Done. Totally agree about consistency and tone.

**Review by m7kvqbe1 — APPROVED** [review]

**Pull request merged**